nptl/sem_destroy.c

```diff
@@ -20,16 +20,22 @@
 #include <semaphore.h>
 #include <shlib-compat.h>
 #include "semaphoreP.h"
+#include <irt_syscalls.h>
 
 
 int
 __new_sem_destroy (sem)
      sem_t *sem;
 {
-  /* XXX Check for valid parameter.  */
+  unsigned int semptr = (unsigned int) sem;
+  int result = __nacl_irt_sem_destroy(semptr);
+  
+  if (result < 0) {
+      __set_errno (-result);
+      return -1;
+  }
 
-  /* Nothing to do.  */
-  return 0;
+  return result;
```

Comment:
I see from man page, sem_destroy returns 0 if success and -1 if fail, so should the return value still be 0 in this case?